File: templates/backend-python/src/constants/roles.py
"""
User roles and permissions.

All role-related constants go here.
"""

class UserRoles:
    """User role constants."""
    ADMIN = "admin"
    USER = "user"
    MODERATOR = "moderator"


class Permissions:
    """Permission constants."""
    READ_USERS = "read:users"
    WRITE_USERS = "write:users"
    DELETE_USERS = "delete:users"
    ADMIN_ACCESS = "admin:access"

File: monorepo/backend-node/src/constants/roles.ts
/**
 * User roles and permissions.
 * 
 * All role-related constants go here.
 */

export class UserRoles {
  static readonly ADMIN = 'admin';
  static readonly USER = 'user';
  static readonly MODERATOR = 'moderator';
}

export class Permissions {
  static readonly READ_USERS = 'read:users';
  static readonly WRITE_USERS = 'write:users';
  static readonly DELETE_USERS = 'delete:users';
  static readonly ADMIN_ACCESS = 'admin:access';
}
